### PR TITLE
quarto report and model_and_results.py

### PR DESCRIPTION
**Pull request by musiccabin**
quarto report: code finalized - just need to test rendering
model_and_results.py: pls review code. ty!

### CODE REVIEW

File: .bash_history
Comment (Psingh238) [inline .bash_history:1]: I recommend adding this file to .gitignore.

File: .jupyter/lab/workspaces/default-37a8.jupyterlab-workspace
Comment (Psingh238) [inline .jupyter/lab/workspaces/default-37a8.jupyterlab-workspace:1]: This entire folder should be added to a .gitignore.

File: .local/share/jupyter/runtime/jupyter_cookie_secret
Comment (Psingh238) [inline .local/share/jupyter/runtime/jupyter_cookie_secret:1]: Add this folder to .gitignore as well.

File: environment.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ dependencies:
   - ucimlrepo=0.0.7
   - vegafusion=1.6.9
   - vegafusion-python-embed=1.6.9
+  - quarto=1.5.57
```
Comment (Psingh238) [inline environment.yml:27]:
```suggestion
```
Remove this change or untrack this file to avoid a merge conflict as `environment.yml` will be updated in a different pull request.

**Review by Psingh238 — COMMENTED** [review]
Some of the files and folders should be added to .gitignore and untracked using a git rm command. The qmd needs to be updated later on after all scripts are merged to only reference results from the scripts. However, I believe it is fine for now and can be updated later.

**Comment by musiccabin** [comment]
i pushed up the changes with the .gitignore. could you pls check if i've done it correctly? @Psingh238 ty!

File: .gitignore [hunk 2]
```diff
@@ -160,3 +160,6 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 #.idea/
+
+.bash_history
+.jupyter/lab/workspaces/default-37a8.jupyterlab-workspace
```
Comment (Psingh238) [inline .gitignore]:
```suggestion
.jupyter/
```
This should ignore everything inside the .jupyter folder.

File: .gitignore [hunk 3]
```diff
@@ -160,3 +160,6 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 #.idea/
+
```
Comment (Psingh238) [inline .gitignore:163]:
```suggestion
.local/
```
This adds the .local folder as well.

**Review by Psingh238 — COMMENTED** [review]
.gitignore has suggested changes. Try using `git rm -r --cached` on the folders to remove them from being tracked as well. Remember to commit the changes to .gitignore first before running this command. The command removes the folders and its contents from being tracked in the future.

**Comment by musiccabin** [comment]
just pushed up these changes along with some other stuff. @Psingh238 if .gitignore still doesn't look right, pls pull my branch and update it. ty!

File: scripts/data_validation.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.16.4
+#   kernelspec:
+#     display_name: Python [conda env:522]
+#     language: python
+#     name: conda-env-522-py
+# ---
+
```
Comment (sgdkd) [inline scripts/data_validation.py:14]: Nothing in this file. This file can be removed?

File: reports/wine_quality_regressor_report.qmd [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+## Methods
+
+### Data
+
+The dataset used in this project is the Wine Quality dataset from the UCI Machine Learning Repository (@cortez2009) and can be found here: https://archive.ics.uci.edu/dataset/186/wine+quality. These datasets are related to red and white variants of the Portuguese "Vinho Verde" wine. They contains physicochemical properties (e.g., acidity, sugar content, and alcohol) of different wine samples, alongside a sensory score representing the quality of the wine, rated by experts on a scale from 0 to 10. Each row in the dataset represents a wine sample, with the columns detailing 11 physicochemical attributes and the quality score. The classes are ordered and not balanced (e.g. there are many more normal wines than excellent or poor ones).
```
Comment (sgdkd) [inline reports/wine_quality_regressor_report.qmd:67]:
```suggestion
The dataset used in this project is the Wine Quality dataset from the UCI Machine Learning Repository (@cortez2009) and can be found [here](https://archive.ics.uci.edu/dataset/186/wine+quality). These datasets are related to red and white variants of the Portuguese "Vinho Verde" wine. They contains physicochemical properties (e.g., acidity, sugar content, and alcohol) of different wine samples, alongside a sensory score representing the quality of the wine, rated by experts on a scale from 0 to 10. Each row in the dataset represents a wine sample, with the columns detailing 11 physicochemical attributes and the quality score. The classes are ordered and not balanced (e.g. there are many more normal wines than excellent or poor ones).
```

**Comment by Psingh238** [comment]
I tested out the rendering with HTML and it worked and now I will be trying out with the PDF rendering. We still need to remove all those code chunks from the qmd since we now have the scripts almost ready.

**Comment by Psingh238** [comment]
Okay, after the changes to the Dockerfile, the PDF is now also rendering.
